include/seqan3/core/platform.hpp

```diff
@@ -49,6 +49,14 @@
 #   error "SeqAn3 requires C++ Concepts, either vie the TS (flag: -fconcepts) or via C++20 (flag: -std=c++2a / -std=c++20)."
 #endif
 
+#if _GNUC_ < 10
```

Comment:
> I thought for < GCC 10 we use `{expression} -> concept_name<__VA_ARGS__>` since this is the current pre GCC10 style confused

No, pre gcc-10 supported

```c++
{expression} -> bool;
```

>= gcc-10 supports ONLY

```c++
{expression} -> std::same_as<bool>;
```

So in a return type needs to be a `concept` name. Before gcc-10 this `concept` was always `std::convertible_to`. You are able since gcc9 to write the following expression:

```c++
{expression} -> std::same_as<bool>;
```

But it actually has some problems that `std::same_as` can't deal with incomplete types. 

This means we know use

```c++
// for >= gcc10:
{expression} -> concept_name<types...>;
// for < gcc10:
{expression}; requires concept_name<decltype(expression), types...>;
```

> And either way, you said GCC9 is more ismilar to GCC 10?

As I wrote in https://github.com/seqan/seqan3/pull/1866#issuecomment-640303597

> BTW gcc 9 does not support `{expression} -> std::same_as<type>` fully either.

Reply:
thanks for clarifying!